mechanisms/CMakeLists.txt

```diff
@@ -12,11 +12,16 @@ set(mod_srcdir "${CMAKE_CURRENT_SOURCE_DIR}/mod")
 
 set(mech_dir "${CMAKE_CURRENT_SOURCE_DIR}/multicore")
 file(MAKE_DIRECTORY "${mech_dir}")
+if(NMC_VECTORIZE_TARGET STREQUAL "KNL" AND NMC_USE_OPTIMIZED_KERNELS)
```

Comment:
> Both the true and false branches perform set(modcc_target "cpu"). Do you mean to set(modcc_target avx512) if this is true?

This branch should be `set(modcc_target avx512)`. That's a mistake that slipped in when I was comparing the intrinsics vs. the original optimised versions.

> How about we remove the NMC_USE_OPTIMIZED_KERNELS option from CMakeLists, and assume that the user is requesting an optimized build if they request a vectorized back end.

That's a valid assumption I think.